src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -663,6 +665,19 @@ public ConsoleHolder getConsoleHolder() {
         return consoleHolder;
     }
 
+    public ConsoleHolder updateConsoleHolder(int inFd, int outFd) {
+        synchronized (this) {
+            final ConsoleHolder newConsoleHolder = ConsoleHolder.update(consoleHolder, inFd, outFd);
+
+            if (newConsoleHolder != consoleHolder) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
```

Comment:
I think we should just give up on the `@CompilationFinal` of `consoleHolder`.
`getConsoleHolder()` is not really fast-path and allowing to update would make the concurrency guarantees too complex to keep the `@CompilationFinal`.

Reply:
In this case, `@CompilationFinal` is incorrect as other threads wouldn't be invalidated by the transfer and could keep using an arbitrarily old ConsoleHolder as it's a constant in the compiled code.

Reply:
That's fine by me.